packages/test-base/src/jvmTest/kotlin/io/realm/kotlin/test/compiler/FullTextTests.kt

```diff
@@ -0,0 +1,171 @@
+            Float::class to "1.4f",
+            Double::class to "1.4",
+            Decimal128::class to "BsonDecimal128(\"1.4E100\")",
+            String::class to "\"Realm\"",
+            RealmInstant::class to "RealmInstant.from(42, 420)",
+            ObjectId::class to "ObjectId.create()",
+            BsonObjectId::class to "BsonObjectId()",
+            RealmUUID::class to "RealmUUID.random()",
+            ByteArray::class to "byteArrayOf(42)",
+            MutableRealmInt::class to "MutableRealmInt.create(42)",
+            RealmAny::class to "RealmAny(42)"
+        )
+        for (type in allFieldTypes) {
+            // TODO Consider adding verification of compiler errors when marking collection
+            //  types as having an index
+            if (type.collectionType != CollectionType.RLM_COLLECTION_TYPE_NONE) {
+                continue
+            }
+
+            val elementType = type.elementType
+            val default = if (!elementType.nullable) defaults[elementType.classifier]
+                ?: error("unmapped default") else null
+
+            val kotlinLiteral = type.toKotlinLiteral()
+            val result = compileFromSource(
+                plugins = listOf(io.realm.kotlin.compiler.Registrar()),
+                source = SourceFile.kotlin(
+                    "fulltext.kt",
+                    """
+                        import io.realm.kotlin.types.MutableRealmInt
+                        import io.realm.kotlin.types.ObjectId
+                        import io.realm.kotlin.types.RealmAny
+                        import io.realm.kotlin.types.RealmInstant
+                        import io.realm.kotlin.types.RealmObject
+                        import io.realm.kotlin.types.RealmUUID
+                        import io.realm.kotlin.types.annotations.FullText
+                        import io.realm.kotlin.RealmConfiguration
+                        import org.mongodb.kbson.BsonObjectId
+                        import org.mongodb.kbson.BsonDecimal128
+
+                        class A : RealmObject {
+                            @FullText
+                            var fulltextKey: $kotlinLiteral = $default
+                        }
+
+                        val configuration =
+                            RealmConfiguration.create(schema = setOf(A::class))
+                    """.trimIndent()
+                )
+            )
+            if (type.isFullTextSupported) {
+                assertEquals(KotlinCompilation.ExitCode.OK, result.exitCode, result.messages)
+            } else {
+                assertEquals(
+                    KotlinCompilation.ExitCode.COMPILATION_ERROR,
+                    result.exitCode,
+                    type.toString()
+                )
+                assertTrue(result.messages.contains(Regex("Full-text key .* is of type .* but must be of type")), result.messages)
+            }
+        }
+    }
+
+    @Test
+    fun fulltext_and_index_cannot_be_combined() {
```

Comment:
FullText and primary key can be combined?

Reply:
No, that shouldn't be allowed. I have added a check preventing this now.